tests/Zend/Locale/FormatTest.php

```diff
@@ -1108,10 +1113,12 @@ public function testToNumberWithoutFormatWithPrecision()
     public function testCheckDateFormatDoesNotEmitNoticeWhenNoOptionsAreNotProvided()
     {
         try {
-            setlocale(LC_ALL, 'en_US'); // test setup
+            $locale = setlocale(LC_ALL, 'en_US'); // test setup
             Zend_Locale_Format::setOptions(array('date_format' => 'yyyy-MM-dd'));
+            $checkDateFormat = Zend_Locale_Format::checkDateFormat('2011-10-21', array());
+            setlocale(LC_ALL, $locale);
```

Comment:
is `$locale` meant to save the previous value? As per https://www.php.net/manual/en/function.setlocale.php#refsect1-function.setlocale-returnvalues `setlocale` returns "new current value". 

Maybe this trick should be used instead to retrieve the locale before the change https://stackoverflow.com/a/33025911/3729316
```suggestion
            $locale = setlocale(LC_ALL, 0); // read current locale
            setlocale(LC_ALL, 'en_US'); // test setup
            Zend_Locale_Format::setOptions(array('date_format' => 'yyyy-MM-dd'));
            $checkDateFormat = Zend_Locale_Format::checkDateFormat('2011-10-21', array());
            setlocale(LC_ALL, $locale); // restore previous locale
```

Reply:
`setlocale(LC_ALL, 0);` must be used to retrieve the previous value, maybe I missed some code. or messed up while rebasing.